### PR TITLE
fix bug: transactional MD5 hash was treated as blob MD5 hash when downloading a range of blob

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -6424,11 +6424,14 @@ namespace Azure { namespace Storage { namespace Blobs {
           {
             response.Details.HttpHeaders.CacheControl = cache_control__iterator->second;
           }
-          auto content_md5__iterator = httpResponse.GetHeaders().find("content-md5");
-          if (content_md5__iterator != httpResponse.GetHeaders().end())
+          if (http_status_code == 200)
```
Comment (gearama) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:6427]: what happens for non 200 cases ? should we at least log something, it looks like we are swallowing non 200 responses, is that the desired behavior?

File: sdk/storage/azure-storage-blobs/test/ut/block_blob_client_test.cpp [hunk 2]
```diff
@@ -110,30 +110,82 @@ namespace Azure { namespace Storage { namespace Test {
 
   TEST_F(BlockBlobClientTest, DownloadTransactionalHash)
```
Comment (gearama) [inline sdk/storage/azure-storage-blobs/test/ut/block_blob_client_test.cpp:111]: could we add some tests for the non 200 case to make sure we have the desired behavior ?

**Review by gearama — DISMISSED** [review]

File: sdk/storage/azure-storage-blobs/test/ut/block_blob_client_test.cpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/test/ut/block_blob_client_test.cpp:111]: done

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:6427]: For 206 responses, blob hash is returned in `x-ms-blob-content-md5` header, which is parsed at L6436

**Review by antkmsft — APPROVED** [review]

**Pull request merged**